### PR TITLE
Return a Dialog with the same network info for destiantion network as…

### PR DESCRIPTION
**Pull request by sigbjornaib**
… what was configured for the provisioning network, or vice versa

### CODE REVIEW

**Comment by itewk** [comment]
@A-Beck @curt-matthews can one of you take a look at this?

File: Automate/RedHatConsulting_Utilities/Service/Provisioning/DynamicDialogs.class/get_networks_from_dialog.rb [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+#
+# @param destination_provider_index Int    Index in the destination templates list for the provider this dialog is for
+# @param network_purpose            String Network purpose [destination_network|provisioning_network]
+
+#
+# Description: Return a Dialog with the same network info for destiantion network as 
+# what was configured for the provisioning network, or vice versa
+#
+
+@DEBUG = true
```
Comment (A-Beck) [inline Automate/RedHatConsulting_Utilities/Service/Provisioning/DynamicDialogs.class/get_networks_from_dialog.rb]: Could we switch `@DEBUG` to default to false to lower verbosity when it is not needed?

Comment (sigbjornaib) [reply Automate/RedHatConsulting_Utilities/Service/Provisioning/DynamicDialogs.class/get_networks_from_dialog.rb]: Sure, I forgot to make this change before I pushed the commit.

File: Automate/RedHatConsulting_Utilities/Service/Provisioning/DynamicDialogs.class/get_networks_from_dialog.rb [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    dialog_source_name = 'provisioning_network'
+    else
+    dialog_source_name = 'unknown'
+  end
+
+  dialog_source = 'dialog_location'
+  dialog_source += '_'
+  dialog_source += destination_provider_index.to_s
+  dialog_source += '_'
+  dialog_source += dialog_source_name
```
[the remark this answers is not in the document]
Comment (sigbjornaib) [reply Automate/RedHatConsulting_Utilities/Service/Provisioning/DynamicDialogs.class/get_networks_from_dialog.rb]: I suppose this provides the same functionality, and we could debate what is the most readable, but I am not going to. I will update the code.

File: Automate/RedHatConsulting_Utilities/Service/Provisioning/DynamicDialogs.class/get_networks_from_dialog.rb [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+  source_network = $evm.root[dialog_source]
```
[the remark this answers is not in the document]
Comment (sigbjornaib) [reply Automate/RedHatConsulting_Utilities/Service/Provisioning/DynamicDialogs.class/get_networks_from_dialog.rb]: The dialog always returns hidden, however the empty hash is a good thing to have. I will update the PR.

**Comment by sigbjornaib** [comment]
All the requested changes has been implemented. Please review again.

**Comment by itewk** [comment]
@A-Beck FYI

**Comment by A-Beck** [comment]
@sigbjornaib looks good - thank you!

**Pull request merged**